### DISCUSSION TITLE
login not possible after migration

### DISCUSSION THREAD

**Discussion by ralfrupf1976** [nextcloud/all-in-one#2561] [Questions] [answered]
Hi - I migrated my Standard Installation of nextcloud to nextcloud aio. So far everything worked. The problem is:

I can't login, the created admin user does not work, my users from the former installation. I can type in username and password but the are all wrong. I can't see any logs or further information.

bye the way: Could it be because I'm running on a differt domain, any security issues? Original Cloud at cloud.ref-xxxxxx Test Envirement running on meet.ref-xxxxxx

**Comment by szaimen** [comment]
As you maybe saw in the migration docs is definitely recommended to use the same domain for when migrating.

Apart from that are you maybe using ldap as user backend? And if yes, did you make sure that the ldap is still reachable from the container?

**Reply by ralfrupf1976** [reply]
Our server will have a bit of a downtime during the migration. That's why i wanted to make sure that everything works well on a second Installation. I think the problem is the the testdomain.

I'll check it out in a few days.

LDAP ist not in use.

**Comment by ralfrupf1976** [comment]
I tried it again. This time with the correct domain. Still the same problem. Everything runs - but i can't login.
The created admin does not work
The earlier users can't login
I can type in my username and password, hit login and then nextcloud says wrong user or password.

Any idea, no ldap used...

**Comment by szaimen** [comment]
Usually there should be somethin in the logs when login is not possible. The log is by default stored here: /var/lib/docker/volumes/nextcloud_aio_nextcloud/_data/data/nextcloud.log

**Reply by ralfrupf1976** [reply]
Thank you - I found the logs! I just post the last lines. As you can see - there are login issues. I don't know what they mean...

`dex.php/apps/bookmarks/public/rest/v2/lock","message":"Login failed: 'ralf.rupf' (Remote IP: '146.185.1.116')","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/113.0","version":"26.0.2.1","data":{"app":"core"}}
{"reqId":"J6YT1GNkULW1S3m4bnwa","level":2,"time":"2023-06-14T20:34:28+00:00","remoteAddr":"194.230.145.133","user":"--","app":"core","method":"PROPFIND","url":"/remote.php/dav/addressbooks/users/michael.pfenninger/contacts/","message":"Login failed: 'michael.pfenninger' (Remote IP: '194.230.145.133')","userAgent":"DAVx5/4.3.3-gplay (2023/06/01; dav4jvm; okhttp/4.11.0) Android/13","version":"26.0.2.1","data":{"app":"core"}}
{"reqId":"aDXWcBCEUNNV3Fyltsdt","level":2,"time":"2023-06-14T20:34:36+00:00","remoteAddr":"146.185.1.116","user":"--","app":"core","method":"POST","url":"/index.php/apps/bookmarks/public/rest/v2/lock","message":"Login failed: 'ralf.rupf' (Remote IP: '146.185.1.116')","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/113.0","version":"26.0.2.1","data":{"app":"core"}}
{"reqId":"6ux2u3frSDcN7eAnh7UP","level":2,"time":"2023-06-14T20:34:36+00:00","remoteAddr":"146.185.1.116","user":"--","app":"core","method":"DELETE","url":"/index.php/apps/bookmarks/public/rest/v2/lock","message":"Login failed: 'ralf.rupf' (Remote IP: '146.185.1.116')","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/113.0","version":"26.0.2.1","data":{"app":"core"}}
{"reqId":"FBOTBQPxfv9J7Km6pAHg","level":2,"time":"2023-06-14T20:34:52+00:00","remoteAddr":"146.185.1.116","user":"--","app":"no app in context","method":"POST","url":"/login","message":"Login failed: Cloudchef (Remote IP: 146.185.1.116)","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/113.0","version":"26.0.2.1","data":[]}
{"reqId":"GI62udBPTlnkWz8SOerO","level":2,"time":"2023-06-15T17:30:13+00:00","remoteAddr":"","user":"--","app":"appstoreFetcher","method":"","url":"--","message":"Could not connect to appstore: cURL error 6: Could not resolve host: apps.nextcloud.com (see https://curl.haxx.se/libcurl/c/libcurl-errors.html) for https://apps.nextcloud.com/api/v1/apps.json","userAgent":"--","version":"26.0.2.1","data":{"app":"appstoreFetcher"}}
{"reqId":"1QQ1Bdfr55gquLx1XjaW","level":2,"time":"2023-06-15T17:35:55+00:00","remoteAddr":"","user":"--","app":"appstoreFetcher","method":"","url":"--","message":"Could not connect to appstore: cURL error 6: Could not resolve host: apps.nextcloud.com (see https://curl.haxx.se/libcurl/c/libcurl-errors.html) for https://apps.nextcloud.com/api/v1/apps.json","userAgent":"--","version":"26.0.2.1","data":{"app":"appstoreFetcher"}}

**Reply by ralfrupf1976** [reply]
docker exec --user www-data nextcloud-aio-nextcloud php occ user:report
+------------------+-----+
| User Report      |     |
+------------------+-----+
| Database         | 0   |
|                  |     |
| total users      | 0   |
|                  |     |
| user directories | 2   |
| active users     | 116 |
| disabled users   | 0   |

**Reply by ralfrupf1976** [reply]
It seems like the database is broken somehow... I tried to use the occ command to show my users. As you can see there are no user in database, but active 116 - what does that mean?
If I try to create a new user the occ command says "ABORTED" - after executing the command. Can't type in a password for the user.
HELP

**Reply by szaimen** [reply]
Can you restore a backup from before the database got broken?

**Reply by ralfrupf1976** [reply]
I think I found the problem - old Installation is a mysql database with encoding utf8mb4_0900_ai_ci
during the database migration:

chunked query, 2 chunks
 1894/1894 [============================] 100%  1 sec/1 sec
 - oc_calendarobjects_props
chunked query, 45 chunks
 44976/44976 [============================] 100% 14 secs/14 secs
 - oc_comments_read_markers
    0 [>---------------------------] < 1 sec
 - oc_dav_shares
 25/25 [============================] 100% < 1 sec/< 1 sec
 - oc_accounts
 120/120 [============================] 100%  1 sec/1 sec
 - oc_directlink
    0 [>---------------------------] < 1 sec
 - oc_calendar_resources
    0 [>---------------------------] < 1 sec
 - oc_calendar_rooms
    0 [>---------------------------] < 1 sec
 - oc_calendar_invitations
 2/2 [============================] 100% < 1 sec/< 1 sec
 - oc_calendar_resources_md
    0 [>---------------------------] < 1 sec
 - oc_calendar_rooms_md
    0 [>---------------------------] < 1 sec
 - oc_dav_cal_proxy
    0 [>---------------------------] < 1 sec
 - oc_calendar_reminders
 358/358 [============================] 100% < 1 sec/< 1 sec
 - oc_properties
chunked query, 4 chunks
 2700/3068 [========================>---]  88%  1 sec/1 sec
In ExceptionConverter.php line 91:

  An exception occurred while executing a query: SQLSTATE[22021]: Character not in repertoire: 7 ERROR:  invalid byte sequence for encoding "UTF8": 0x00

In Exception.php line 30:

  SQLSTATE[22021]: Character not in repertoire: 7 ERROR:  invalid byte sequence for encoding "UTF8": 0x00

In Statement.php line 101:

  SQLSTATE[22021]: Character not in repertoire: 7 ERROR:  invalid byte sequence for encoding "UTF8": 0x00

**Reply by szaimen** [reply]
Yes, looks like it

**Reply by ralfrupf1976** [reply]
Any idea how to fix it?

**Reply by szaimen** [reply]
I've found this article: https://confluence.atlassian.com/stashkb/database-migration-to-postgresql-fails-invalid-byte-sequence-for-encoding-utf8-0x00-696386363.html

**Reply by szaimen** [reply]
So apparently you will need to update all columns type text field that are set to 0x00 to an empty string before running the migration. In best case actually the occ migrate command should do this so I would suggest to open an issue here: https://github.com/nextcloud/server/issues

**Reply by szaimen** [reply, marked as answer]
Actually there is this issue already: https://github.com/nextcloud/server/issues/37754

**Reply by ralfrupf1976** [reply]
Yes, found it too! Thanks for your help!!
So, i have to wait - too big for my nextcloud knowlegde..

**Reply by ralfrupf1976** [reply]
Hoping that the fix will come with nextcloud 26.0.3 - next week. Otherwise I can't switch to nextcloud aio :-(

**Reply by ralfrupf1976** [reply]
I just updated to Version 26.0.3 - still the same issue when trying to convert the database :-(

 - oc_properties
chunked query, 4 chunks
 2800/3068 [=========================>--]  91%  1 sec/1 sec
In ExceptionConverter.php line 91:

  An exception occurred while executing a query: SQLSTATE[22021]: Character not in repertoire: 7 ERROR:  invalid byte sequence for encoding "UTF8": 0x00

In Exception.php line 30:

  SQLSTATE[22021]: Character not in repertoire: 7 ERROR:  invalid byte sequence for encoding "UTF8": 0x00

In Statement.php line 101:

  SQLSTATE[22021]: Character not in repertoire: 7 ERROR:  invalid byte sequence for encoding "UTF8": 0x00

**Reply by szaimen** [reply]
yes, the issue is not closed yet. You can subscribe to it.
